airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/ctid/CtidGlobalStateManager.java

```diff
@@ -50,23 +51,29 @@ public CtidGlobalStateManager(final CtidStreams ctidStreams,
     this.stateManager = stateManager;
     this.savedOffsetAfterReplicationSlotLSN = savedOffsetAfterReplicationSlotLSN;
     this.defaultCdcState = defaultCdcState;
-    initStream(ctidStreams, catalog);
     this.fileNodeHandler = fileNodeHandler;
+    initStream(ctidStreams, catalog);
   }
 
   private void initStream(final CtidStreams ctidStreams,
                           final ConfiguredAirbyteCatalog catalog) {
     this.streamsThatHaveCompletedSnapshot = new HashSet<>();
     this.resumableFullRefreshStreams = new HashSet<>();
+    this.nonResumableFullRefreshStreams = new HashSet<>();
     catalog.getStreams().forEach(configuredAirbyteStream -> {
+      var pair =
+          new AirbyteStreamNameNamespacePair(configuredAirbyteStream.getStream().getName(), configuredAirbyteStream.getStream().getNamespace());
       if (!ctidStreams.streamsForCtidSync().contains(configuredAirbyteStream) && configuredAirbyteStream.getSyncMode() == SyncMode.INCREMENTAL) {
-        streamsThatHaveCompletedSnapshot.add(
-            new AirbyteStreamNameNamespacePair(configuredAirbyteStream.getStream().getName(), configuredAirbyteStream.getStream().getNamespace()));
+        streamsThatHaveCompletedSnapshot.add(pair);
       }
-      if (ctidStreams.streamsForCtidSync().contains(configuredAirbyteStream)
-          && configuredAirbyteStream.getSyncMode() == SyncMode.FULL_REFRESH) {
-        this.resumableFullRefreshStreams.add(
-            new AirbyteStreamNameNamespacePair(configuredAirbyteStream.getStream().getName(), configuredAirbyteStream.getStream().getNamespace()));
+
+      if (configuredAirbyteStream.getSyncMode() == SyncMode.FULL_REFRESH) {
+        if (fileNodeHandler.hasFileNode(
+            new AirbyteStreamNameNamespacePair(configuredAirbyteStream.getStream().getName(), configuredAirbyteStream.getStream().getNamespace()))) {
+          this.resumableFullRefreshStreams.add(pair);
+        } else {
+          this.nonResumableFullRefreshStreams.add(pair);
```

Comment:
Can we add logs/analytics messages for number of non resumable full refresh streams? I'd be curious how often they are encountered

Reply:
Sure - I'm curious how do we do that?
Alternatively I guess we can also query platform database (config-db) to see how many streams has that is_resumable flag on catalog

Reply:
Here is an example of how to emit it: https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialReadUtil.java#L153

You need to define an analytics message in this file : https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/DbAnalyticsUtils.kt

Once that's in, I can show you how to create a dashboard from this!

Reply:
Not blocking - I'll leave it up to you on what you think works since as you pointed out, this information is also available. by querying platform DB

Reply:
Got it, thanks for the pointer! 

I think maybe querying db fits this more. We can know who run them and also how often they are running